src/Orleans/Core/Exceptions.cs

```diff
@@ -170,6 +170,21 @@ protected GrainExtensionNotInstalledException(SerializationInfo info, StreamingC
     }
 
     /// <summary>
+    /// Signifies that an request was cancelled due to target silo death.
+    /// </summary>
+    [Serializable]
+    public class SilosDeathException : OrleansException
```

Comment:
How about `SiloUnavailableException`? It doesn't presume whether or not the silo has become permanently or temporarily unavailable.